src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -0,0 +1,542 @@
+        } else {
+            // If updating an existing entity, specify record to update fields found with where clause on id.
+            statementString = String.format("update %s set %s where id = %d", tableName, String.join(", ", fieldNames), id);
+        }
+        // Prepare statement and set RETURN_GENERATED_KEYS to get the entity id back
+        PreparedStatement preparedStatement = connection.prepareStatement(
+                statementString,
+                Statement.RETURN_GENERATED_KEYS);
+        if (!batch) {
+            fillValues(fieldStringMap, preparedStatement, connection);
+        }
+        return preparedStatement;
+    }
+
+    private static void fillValues(Map<Field, String> fieldStringMap, PreparedStatement preparedStatement, Connection connection) throws SQLException {
+        // one-based index for statement parameters
+        int index = 1;
+        // Sort keys on field name to match prepared statement parameters
+        Field[] keys = fieldStringMap.keySet().stream()
+                .sorted(Comparator.comparing(Field::getName))
+                .toArray(Field[]::new);
+        // Fill statement parameters with strings from JSON
+        for (Field field : keys) {
+            String value = fieldStringMap.get(field);
+//            LOG.info("{}={}", field.name, value);
+            try {
+                if (value == null) {
+                    // Handle null value
+                    preparedStatement.setNull(index, field.getSqlType().getVendorTypeNumber());
+                } else {
+                    field.setParameter(preparedStatement, index, value);
+                }
+            } catch (StorageException e) {
+                if (field.name.contains("_time")) {
+                    // FIXME: This is a hack to get arrival and departure time into the right format. Because the UI
+                    // currently returns them as seconds since midnight rather than the Field-defined format HH:MM:SS.
+                    try {
+                        if (value == null) {
+                            preparedStatement.setNull(index, field.getSqlType().getVendorTypeNumber());
+                        } else {
+                            Integer seconds = Integer.parseInt(value);
+                            preparedStatement.setInt(index, seconds);
+                        }
+                    } catch (NumberFormatException ex) {
+                        connection.rollback();
+                        LOG.error("Bad column: {}={}", field.name, value);
+                        ex.printStackTrace();
+                        throw ex;
+                    }
+                } else {
+                    connection.rollback();
+                    throw e;
+                }
+            }
+            index += 1;
+        }
+    }
+
+    /**
+     * This updates
```

Comment:
Please expand Javadoc. Generally I'm having trouble following what is meant by parent/child tables, referencing tables, etc. Although I can mostly guess, I'm missing a definitive picture of what these things mean.

Reply:
Yes, I think I was having a bit of a hard time keeping that straight too.  The model I set up breaks down with the pattern -> shape relationship.  In most cases the "child table" should have a many-to-one reference back to the parent (`stop_times` has a `trip_id` reference to `trips` table).  However, with the pattern to shapes relationship, the foreign key (shape_id) is on the patterns table rather than shapes.

Let's discuss/work on this together to come up with something better/more coherent.